// plugin-welcome/src/api/models/EntUserEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntConfirmation,
    EntConfirmationFromJSON,
    EntConfirmationFromJSONTyped,
    EntConfirmationToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntUserEdges
 */
export interface EntUserEdges {
    /**
     * UserConfirmation holds the value of the UserConfirmation edge.
     * @type {Array<EntConfirmation>}
     * @memberof EntUserEdges
     */
    userConfirmation?: Array<EntConfirmation>;
}

export function EntUserEdgesFromJSON(json: any): EntUserEdges {
    return EntUserEdgesFromJSONTyped(json, false);
}

export function EntUserEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntUserEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'userConfirmation': !exists(json, 'userConfirmation') ? undefined : ((json['userConfirmation'] as Array<any>).map(EntConfirmationFromJSON)),
    };
}

export function EntUserEdgesToJSON(value?: EntUserEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'userConfirmation': value.userConfirmation === undefined ? undefined : ((value.userConfirmation as Array<any>).map(EntConfirmationToJSON)),
    };
}
